tests/unit/src/test/scala/tests/LineListenerSuite.scala

```diff
@@ -0,0 +1,63 @@
+package tests
+
+import scala.meta.internal.metals.LineListener
+import scala.collection.mutable
+
+object LineListenerSuite extends BaseSuite {
```

Comment:
Couldn't hurt to check how it behaves with unicode now, when we operate on bytes.

Reply:
The state machine operates on characters and does not nothing until it hits the first character with int value 27. I've added a new test case to demonstrate that unicode characters are unaffected.